tests/roxctl/bats-tests/cluster/sensor-generate-image-overrides.bats

```diff
@@ -33,68 +33,38 @@ registry_from_flavor() {
   esac
 }
 
-collector_full_from_flavor() {
-   case "$central_flavor" in
-   "development_build")
-     echo "collector:$any_version_latest"
-     ;;
-   "opensource")
-     echo "collector:$any_version"
-     ;;
-   esac
-}
-
-collector_slim_from_flavor() {
-    case "$central_flavor" in
-     "development_build")
-       echo "collector:$any_version_slim"
-       ;;
-     "opensource")
-       echo "collector-slim:$any_version"
-       ;;
-     esac
-}
-
 any_version_latest="${any_version}[0-9]+\-latest"
 any_version_slim="${any_version}[0-9]+\-slim"
 
 @test "roxctl sensor generate: no overrides" {
   generate_bundle k8s --name "$cluster_name"
   assert_success
   assert_bundle_registry "$out_dir" "sensor" "$(registry_from_flavor)/main:$any_version"
-  assert_bundle_registry "$out_dir" "collector" "$(registry_from_flavor)/$(collector_slim_from_flavor)"
-  delete_cluster "$cluster_name"
-}
-
-@test "roxctl sensor generate: no overrides with collector full" {
```

Comment:
(code line unrelated) We remove the e2e test cases for roxctl here, but we still want roxctl to work correctly with older centrals. Could we maybe add a test for that?
We could call the roxctl with params for collector slim and make sure that the error is as expected (or there is no error). I can imagine that removing all collector slim from roxctl would produce a different error than requesting bundle from central that does not support collector slim.

Reply:
Good point, I'll add a test here, but I think all we can do is check for success and lack of -slim suffix since these tests run against the same version of Central. 

Do we have other roxctl tests for version mismatches elsewhere?

Reply:
I am not sure - maybe there are some used in e2e test where we use roxctl to deploy stuff? I don't think their goal is to test version mismatches though.

Even in bats, we may not have them at the moment. The path `/bats-tests/cluster` means that we need a running Central to execute those, but I don't remember adding bats tests for roxctl with different versions of Central (that is what I remember from 2022 when we worked on the flavors feature).

Adding a test that would deploy central in older version and run a bats suite may be an overkill for this PR, so I am fine to tackle this in a followup. But maybe we could add a test in `/bats-tests/local` to check that providing the slim option does not trigger any error similar to `unknown parameter --slim`?

Reply:
I've been thinking about this local test and I think if I add an `assert_line ...` for the deprecation warning to the new test here, then we can cover this behaviour. 

I'm not sure it makes sense to test this command (sensor generate) without some cluster there, so perhaps this would be a good middle ground until we can look at adding a test against an older central. WDYT?

Reply:
Yes, that is fine to use a local test and assert on a line.

Reply:
Done

Reply:
Or, to be more precise, _now_ it's done :sweat_smile: